dotCMS/src/main/java/com/dotcms/publisher/ajax/RemotePublishAjaxAction.java

```diff
@@ -351,18 +352,39 @@ public void retry ( HttpServletRequest request, HttpServletResponse response ) t
                 auditHistory.setNumTries( 0 );
                 publishAuditAPI.updatePublishAuditStatus( configStatic.getId(), status.getStatus(), auditHistory, true );
 
+                Publisher staticPublisher;
+
                 try{
-                    awss3Publisher.init(configStatic);
-                    awss3Publisher.process(null);
+
+                    List<Environment> environments = APILocator.getEnvironmentAPI().findEnvironmentsByBundleId(bundleId);
+
+                    for (Environment environment : environments) {
+
+                        List<PublishingEndPoint> endPoints = APILocator.getPublisherEndPointAPI().findSendingEndPointsByEnvironment(environment.getId());
+                        PublishingEndPoint targetEndpoint = endPoints.get(0);
+
+                        //Processing AWS push retry
+                        if (AWSS3Publisher.PROTOCOL_AWS_S3.equalsIgnoreCase(targetEndpoint.getProtocol())){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)